experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs

```diff
@@ -0,0 +1,132 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json.Linq;
+
+namespace LivePersonAgentBot.Bots
+{
+    public class LivePersonAgentBot : ActivityHandler
+    {
+        private readonly BotState _conversationState;
+
+        public LivePersonAgentBot(ConversationState conversationState)
+        {
+            _conversationState = conversationState;
+        }
+
+        static Dictionary<string, string> Capitals = new Dictionary<string, string>
+        {
+            ["France"] = "Paris",
+            ["Italy"] = "Rome",
+            ["Japan"] = "Tokyo",
+            ["Poland"] = "Warsaw",
+            ["Germany"] = "Hamburg" // not really
+        };
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var conversationStateAccessors = _conversationState.CreateProperty<LoggingConversationData>(nameof(LoggingConversationData));
+            var conversationData = await conversationStateAccessors.GetAsync(turnContext, () => new LoggingConversationData());
+
+            var userText = turnContext.Activity.Text.ToLower();
+            if (userText.Contains("agent"))
+            {
+                await turnContext.SendActivityAsync("Your request will be escalated to a human agent");
+
+                var transcript = new Transcript(conversationData.ConversationLog.Where(a => a.Type == ActivityTypes.Message).ToList());
+
+                var evnt = EventFactory.CreateHandoffInitiation(turnContext,
+                    new { Skill = "credit-cards" },
+                    transcript);
+
+                await turnContext.SendActivityAsync(evnt);
+            }
+            else
+            {
+                string replyText = $"Sorry, I cannot help you.";
+                if (userText == "hi")
+                {
+                    replyText = "Hello!";
+                }
+                if (userText.Contains("disneyland"))
+                {
+                    await turnContext.SendActivityAsync("Sorry I cannot answer your questoin but an agent will be with you shortly");
+
+                    var channelData = @"
+  {
+    ""action"": {
+        ""name"": ""TRANSFER"",
+        ""parameters"": {
+          ""skill"": ""Fallback-Skill""
+        }
+    }
+  }";
+                    var activity = new Activity("message");
+                    activity.Text = "";
+                    activity.ChannelData = JObject.Parse(channelData);
+
+                    await turnContext.SendActivityAsync(activity);
+                    return;
+                }
+                else
+                {
+                    foreach (var country in Capitals.Keys)
+                    {
+                        if (userText.Contains(country.ToLower()))
```

Comment:
Do you want a culture aware ToLower()? I don't think the thread culture is ever being set, so I suspect not.